src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -759,7 +767,30 @@ private MenuBar createMenu() {
                 factory.createMenuItem(StandardActions.SET_FILE_LINKS, new AutoLinkFilesAction(this, prefs, stateManager, undoManager))
         );
 
-        final PushToApplicationAction pushToApplicationAction = new PushToApplicationAction(stateManager, this.getPushApplications(), this.getDialogService());
+        // PushToApplication submenu
+        final Menu pushApplicationsMenu = factory.createSubMenu(StandardActions.SELECT_PUSH_APPLICATION);
+        final String activePushToApplicationName = Globals.prefs.get(JabRefPreferences.PUSH_TO_APPLICATION);
+
+        final ToggleGroup pushToToggleGroup = new ToggleGroup();
+        RadioMenuItem pushToApplication;
```

Comment:
Please move the declaration of `pushToApplication` and `pushToApplicationMenuAction` down to the body of the for loop (because otherwise one would expect that you need these variables also after the for-loop).

Reply:
done